fast_pred/src/fastlightfm.cpp

```diff
@@ -190,15 +191,28 @@ void FastLightFM::load(string dir) {
   user_features = CSRMatrix::newInstance(cnpy::npz_load(dir + "/user-features.npz"));
   item_features = CSRMatrix::newInstance(cnpy::npz_load(dir + "/item-features.npz"));
 
+  init();
+}
+
+void FastLightFM::init() {
+
   assert(item_embeddings.shape[1] == user_embeddings.shape[1]);
   no_components = item_embeddings.shape[1];
+
+  number_of_items = item_features->rows;
+  number_of_users = user_features->rows;
+
+  int stride = no_components + 1;
+  lightfm_cache = new FastLightFMCache(number_of_items, number_of_users, stride);
```

Comment:
Might need to clean this up in the destructor?

Reply:
yeah, we should have delete lightfm_cache in  the destructor